x-pack/plugins/observability/server/services/slo/resource_installer.ts

```diff
@@ -138,16 +154,20 @@ export class DefaultResourceInstaller implements ResourceInstaller {
 
   private async createOrUpdateComponentTemplate(template: ClusterPutComponentTemplateRequest) {
     this.logger.debug(`Installing SLO component template ${template.name}`);
-    return this.esClient.cluster.putComponentTemplate(template);
+    return this.execute(() => this.esClient.cluster.putComponentTemplate(template));
   }
 
   private async createOrUpdateIndexTemplate(template: IndicesPutIndexTemplateRequest) {
     this.logger.debug(`Installing SLO index template ${template.name}`);
-    return this.esClient.indices.putIndexTemplate(template);
+    return this.execute(() => this.esClient.indices.putIndexTemplate(template));
   }
 
   private async createOrUpdateIngestPipelineTemplate(template: IngestPutPipelineRequest) {
     this.logger.debug(`Installing SLO ingest pipeline template ${template.id}`);
-    await this.esClient.ingest.putPipeline(template);
+    return this.execute(() => this.esClient.ingest.putPipeline(template));
+  }
+
+  private async execute<T>(esCall: () => Promise<T>): Promise<T> {
+    return await retryTransientEsErrors(esCall, { logger: this.logger });
```

Comment:
Interesting that this is needed!

Reply:
Well, locally it works fine without it, but I want to be sure we end up installing the resources even when the es node is temporarily not responding